sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -198,29 +202,62 @@ public virtual Response<ConfigurationSetting> Set(string key, string value, stri
         {
             if (string.IsNullOrEmpty(key))
                 throw new ArgumentNullException($"{nameof(key)}");
-            return Set(new ConfigurationSetting(key, value, label), cancellationToken);
+            return Set(new ConfigurationSetting(key, value, label), default(HttpRequestOptions), cancellationToken);
         }
 
         /// <summary>
         /// Creates a <see cref="ConfigurationSetting"/> if it doesn't exist or overrides an existing setting in the configuration store.
         /// </summary>
         /// <param name="setting"><see cref="ConfigurationSetting"/> to create.</param>
+        /// <param name="onlyIfUnchanged"></param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        public virtual async Task<Response<ConfigurationSetting>> SetAsync(ConfigurationSetting setting, CancellationToken cancellationToken = default)
+        public virtual async Task<Response<ConfigurationSetting>> SetAsync(ConfigurationSetting setting, bool onlyIfUnchanged = false, CancellationToken cancellationToken = default)
+        {
+            if (setting == null)
+                throw new ArgumentNullException($"{nameof(setting)}");
```

Comment:
minor: throw new ArgumentNullException(nameof(setting));

(no format necessary)

Reply:
Will address in https://github.com/Azure/azure-sdk-for-net/issues/7732